src/client/Microsoft.Identity.Client/ApiConfig/SystemWebViewOptions.cs

```diff
@@ -1,169 +1,176 @@
-        /// On Linux, uses the default system browser instead, as Edge is not available.
-        /// </summary>
-        public static async Task OpenWithChromeEdgeBrowserAsync(Uri uri)
-        {
-            if (uri == null)
-            {
-                throw new ArgumentNullException(nameof(uri));
-            }
-
-            string url = uri.AbsoluteUri;
-
-#if DESKTOP
-            url = url.Replace("&", "^&");
-            Process.Start(new ProcessStartInfo("cmd", $"/c start msedge {url}") { CreateNoWindow = true });
-            await Task.FromResult(0).ConfigureAwait(false);
-#else
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                url = url.Replace("&", "^&");
-                Process.Start(new ProcessStartInfo("cmd", $"/c start msedge {url}") { CreateNoWindow = true });
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-            {
-                var proxy = PlatformProxyFactory.CreatePlatformProxy(new NullLogger());
-                await proxy.StartDefaultOsBrowserAsync(url).ConfigureAwait(false);
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
-            {
-                Process.Start("msedge", url);
-            }
-            else
-            {
-                throw new PlatformNotSupportedException(RuntimeInformation.OSDescription);
-            }
-#endif
-
-        }
-#endif
-    }
-}
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Logger;
+using Microsoft.Identity.Client.PlatformsCommon.Factories;
+
+namespace Microsoft.Identity.Client
+{
+    // Default browser WebUI is not available on mobile (Android, iOS, UWP), but allow it at runtime
+    // to avoid MissingMethodException
+
+    /// <summary>
+    /// Options for using the default OS browser as a separate process to handle interactive auth.
+    /// MSAL will be listening for the OS browser to finish authenticating, but it cannot close the browser.
+    /// It can however respond with a 200 OK message or a 302 Redirect, which can be configured here.
+    /// For more details see https://aka.ms/msal-net-os-browser
+    /// </summary>
+#if !SUPPORTS_OS_SYSTEM_BROWSER
+    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+#endif
+    public class SystemWebViewOptions
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public SystemWebViewOptions()
+        {
+            ValidatePlatformAvailability();
+        }
+
+        /// <summary>
+        /// When the user finishes authenticating, MSAL will respond with a 200 OK message,

```

Comment:
Please add comments that these existing options are exclusive for desktop operating systems (so .NET Clasic and .NET Core). And if the user tries to `set` one of these on Android, iOS or UWP, we should throw a nice exception.

Reply:
Added the hide attribute